addons/misra.py

```diff
@@ -1107,11 +1150,17 @@ def misra_13_6(data):
 
 def misra_14_1(data):
     for token in data.tokenlist:
-        if token.str != 'for':
-            continue
-        exprs = getForLoopExpressions(token)
-        if exprs and hasFloatComparison(exprs[1]):
-            reportError(token, 14, 1)
+        if token.str != 'for' and token.str != 'while':
+            continue
+        if token.str == 'for':
+            exprs = getForLoopExpressions(token)
+            for counter in findCounterTokens(exprs[1]):
```

Comment:
It would be a good idea to check that `exprs` is not None, wouldn't it? also is it best to check that `len(exprs)` is 3 or is that implied when exprs is not None?

Reply:
>It would be a good idea to check that exprs is not None, wouldn't it?

Yes You are right. 

>also is it best to check that len(exprs) is 3 or is that implied when exprs is not None?

Yes .. according to me.. if exprs is not None.. len(exprs) is 3